client/src/client.rs

```diff
@@ -547,8 +552,10 @@ where
         event::wait(
             &events,
             |e| match e {
-                Event::PeerManager(peermgr::Event::PeerNegotiated { addr }) => {
-                    negotiated.insert(addr);
+                Event::PeerManager(peermgr::Event::PeerNegotiated { addr, services }) => {
+                    if services == supported_services.into() {
```

Comment:
This won't work, we want to check that the required services are included in the services provided by the peer, not that they are an exact match.

Reply:
Fixed, thanks.